services/packagist/packagist-dependency-version.tester.js

```diff
@@ -0,0 +1,88 @@
+import { isComposerVersion } from '../test-validators.js'
+import { createServiceTester } from '../tester.js'
+export const t = await createServiceTester()
+
+t.create('gets the package version of symfony 5.2.3 - without query params')
+  .get('/symfony/symfony/v5.2.3.json')
+  .expectBadge({
+    label: 'dependency version',
+    message: 'version requirement not found',
+  })
+
+t.create('gets the package version of symfony - with correct dependency')
+  .get('/symfony/symfony/twig/twig.json')
+  .expectBadge({ label: 'twig/twig', message: isComposerVersion })
+
+t.create(
+  'gets the package version of symfony - with correct dependency and specified version'
+)
+  .get('/symfony/symfony/twig/twig.json?version=v3.2.8')
+  .expectBadge({ label: 'twig/twig', message: isComposerVersion })
+
+t.create(
+  'gets the package version of symfony - with correct dependency, correct version and correct custom server'
+)
+  .get(
+    '/symfony/symfony/twig/twig.json?version=v3.2.8&server=https://packagist.org'
+  )
+  .expectBadge({ label: 'twig/twig', message: isComposerVersion })
+
+t.create(
+  'gets the package version of symfony - with correct dependency and specified version and incorrect custom server'
+)
+  .get(
+    '/symfony/symfony/twig/twig.json?version=v3.2.8&server=https://packagisttttttt.org'
+  )
+  .expectBadge({
+    label: 'dependency version',
+    message: 'inaccessible',
+  })
+
+t.create('gets the package version of symfony - with incorrect dependency name')
+  .get('/symfony/symfony/twig/twiiiiiiig.json?version=v3.2.8')
+  .expectBadge({
+    label: 'dependency version',
+    message: 'version requirement not found',
+  })
+
+t.create('gets the package version of symfony - with incorrect version')
+  .get('/symfony/symfony/twig/twig.json?version=v3.2.80000')
+  .expectBadge({
+    label: 'dependency version',
+    message: 'invalid version',
+  })
+
+t.create(
+  `gets the package version of symfony 5.2.3 - incorrectly with missing part of dependency's name`
+)
+  .get('/symfony/symfony/twig.json')
+  .expectBadge({
+    label: 'dependency version',
+    message: 'version requirement not found',
+  })
```

Comment:
In general, the naming of tests in this file is a bit jumbled. For example, this one is called `gets the package version of symfony 5.2.3...` but you're not passing `?version=5.2.3`. There's another one further down with the same problem that I've suggested deleting as a duplicate. Then you've got another one further up where you're actually explicitly passing `?version=v3.2.8` in the URL but the test name is just "gets the package version of symfony...", and so on.

Can you have a look over the tests and make sure:

- you're passing the `?version=` param when that is actually part of the test, and leave it out if it doesn't make any difference to the test
- if you want to explicitly describe the test params in the name of the test, make sure the test names and the params in the test body match up consistently.

That said, I know the number of cases we need to cover is now a bit larger and naming things is hard, but if you look at the test names that are deleted elsewhere in this PR (e.g: "invalid package name", "invalid version", "custom server", "invalid custom server", etc) those test names are actually pretty good. They explain what we are actually trying to test (i.e: what behaviour are we trying to trigger by providing these inputs, and what are we making an assertion about) rather than describing all of the inputs in each test. You might get more mileage out of just simplifying the test names a bit. I'll leave it with you to have a think about it.

Reply:
I removed the `version` query param, where it was not needed.
I modified the tests' names, I hope that now they are ok.